ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -760,6 +786,66 @@ doAllocBuffers(unsigned bufferSize,
     rstKeywords["PRES"] = 0;
     rstKeywords["PRESSURE"] = 0;
 
+    if (enableMech_ && eclState_.runspec().mech()) {
+        this->mechPotentialForce_.resize(bufferSize,0.0);
+        rstKeywords["MECHPOTF"] = 0;
+        this->mechPotentialTempForce_.resize(bufferSize,0.0);
+        rstKeywords["TEMPPOTF"] = 0;
+        this->mechPotentialPressForce_.resize(bufferSize,0.0);
+        rstKeywords["PRESPOTF"] = 0;
```

Comment:
I'll repeat [my comment from earlier](https://github.com/OPM/opm-simulators/pull/4588#discussion_r1291260147):

> If `enableMech_` is sufficient to request that geomechanical result arrays be output and we do not expect that users will have something along the lines of
> ```
> RPTRST
>   MECHPOTF TEMPPOTF PRESPOTF ... /
> ```
> in their input decks, then we do not need these `rstKeywords["something"] = 0` statements in here.

The question then becomes: Do we expect that users will enable individual result arrays in `RPTRST`?  Because if we do, then we also need to [tell the output system about them](https://github.com/OPM/opm-common/blob/e22b0944ddce07c58515dcb9b127d504eba995d1/src/opm/input/eclipse/Schedule/RSTConfig.cpp#L162).

Reply:
I think writing out everything is ok for now. In other cases I think negating output is better.